code/modules/reagents/reagent_containers/iv_bag.dm

```diff
@@ -0,0 +1,179 @@
+		return
+
+	if(get_dist(get_turf(src), get_turf(injection_target)) > 1)
+		to_chat(injection_target, "<span class='userdanger'>The [src]'s' needle is ripped out of you!</span>")
+		injection_target.apply_damage(3, BRUTE, pick("r_arm", "l_arm"))
+		end_processing()
+		return
+
+	if(mode) 	// Injecting
+		if(reagents.total_volume)
+			var/fraction = min(amount_per_transfer_from_this/reagents.total_volume, 1) 	//The amount of reagents we'll transfer to the person
+			reagents.reaction(injection_target, INGEST, fraction) 						//React the amount we're transfering.
+			reagents.trans_to(injection_target, amount_per_transfer_from_this)
+			update_icon()
+	else		// Drawing
+		if(reagents.total_volume < reagents.maximum_volume)
+			injection_target.transfer_blood_to(src, amount_per_transfer_from_this)
+			for(var/datum/reagent/x in injection_target.reagents.reagent_list) // Pull small amounts of reagents from the person while drawing blood
+				injection_target.reagents.trans_to(src, amount_per_transfer_from_this/10)
+			update_icon()
+
+/obj/item/reagent_containers/iv_bag/attack(mob/living/M, mob/living/user, def_zone)
+	return
+
+/obj/item/reagent_containers/iv_bag/afterattack(atom/target, mob/user, proximity)
+	if(!proximity)
+		return
+	if(!target.reagents)
+		return
+
+	if(isliving(target))
+		var/mob/living/L = target
+		if(injection_target) // Removing the needle
+			if(L != injection_target)
+				to_chat(user, "<span class='notice'>[src] is already inserted into [injection_target]'s arm!")
+				return
+			if(L != user)
+				L.visible_message("<span class='danger'>[user] is trying to remove [src]'s needle from [L]'s arm!</span>", \
+								"<span class='userdanger'>[user] is trying to remove [src]'s needle from [L]'s arm!</span>")
+				if(!do_mob(user, L))
+					return
+			L.visible_message("<span class='danger'>[user] removes [src]'s needle from [L]'s arm!</span>", \
+								"<span class='userdanger'>[user] removes [src]'s needle from [L]'s arm!</span>")
+			end_processing()
+		else // Inserting the needle
+			if(!L.can_inject(user, 1))
+				return
+			if(L != user)
+				L.visible_message("<span class='danger'>[user] is trying to insert [src]'s needle into [L]'s arm!</span>", \
+									"<span class='userdanger'>[user] is trying to insert [src]'s needle into [L]'s arm!</span>")
+				if(!do_mob(user, L))
+					return
+			L.visible_message("<span class='danger'>[user] inserts [src]'s needle into [L]'s arm!</span>", \
+									"<span class='userdanger'>[user] inserts [src]'s needle into [L]'s arm!</span>")
+			begin_processing(L)
+
+
+/obj/item/reagent_containers/iv_bag/update_icon()
+	overlays.Cut()
+
+	if(reagents.total_volume)
+		var/percent = round((reagents.total_volume / volume) * 10) // We round the 1's place off of our percent for easy image processing.
+		var/image/filling = image('icons/goonstation/objects/iv.dmi', src, "[icon_state][percent]")
+
+		filling.icon += mix_color_from_reagents(reagents.reagent_list)
+		overlays += filling
```

Comment:
You used `cut_overlays` and `add_overlay` for the defib mount, why not this?

Reply:
Because the defib mount is a /tg/ port and the procs never reported any errors. Overlays behaved as expected. The bags are my original code.